internal/raft/cluster.go

```diff
@@ -485,23 +493,24 @@ func (c *Cluster) withRetry(ctx context.Context, shardID, replicaID uint64, f fu
 
 		err := f(ctx)
 		duration := retry.Duration()
-		if errors.Is(err, dragonboat.ErrShardNotReady) {
-			logger.Debugf("shard not ready, retrying in %s", duration)
-			time.Sleep(duration)
-			if _, ok := <-ctx.Done(); ok {
-				return fmt.Errorf("context cancelled")
+
+		if err != nil {
+			retried := false
+			for _, retryError := range retryErrors {
+				if errors.Is(err, retryError) {
+					logger.Debugf("got error %s, retrying in %s", err, duration)
```

Comment:
Our convention is that logs should be capitalised...here and elsewhere

Reply:
Changing. Also, sounds like a job for a linter